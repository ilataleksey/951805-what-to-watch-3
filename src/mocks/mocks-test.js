export const films = [
  {
    name: `Наименование фильма-1`,
    poster: `img/film-poster.jpg`,
    preview: `video/film.mp4`,
    genre: `genre-1`
  },
  {
    name: `Наименование фильма-2`,
    poster: `img/film-poster.jpg`,
    preview: `video/film.mp4`,
    genre: `genre-1`
  },
  {
    name: `Наименование фильма-3`,
    poster: `img/film-poster.jpg`,
    preview: `video/film.mp4`,
    genre: `genre-1`
  },
  {
    name: `Наименование фильма-4`,
    poster: `img/film-poster.jpg`,
    preview: `video/film.mp4`,
    genre: `genre-2`
  },
  {
    name: `Наименование фильма-5`,
    poster: `img/film-poster.jpg`,
    preview: `video/film.mp4`,
    genre: `genre-1`
  },
  {
    name: `Наименование фильма-6`,
    poster: `img/film-poster.jpg`,
    preview: `video/film.mp4`,
    genre: `genre-2`
  },
];

export const film = {
  name: `Наименование фильма`,
  poster: `img/film-poster.jpg`,
  preview: `video/film.mp4`,
  genre: `genre-1`
};

export const activeGenre = `genre-2`;

export const genresList = [`genre-1`, `genre-2`, `genre-3`, `genre-4`];
