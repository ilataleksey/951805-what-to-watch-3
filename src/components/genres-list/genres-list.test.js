import React from "react";
import renderer from "react-test-renderer";
import GenresList from "./genres-list.jsx";
import {films, genresList} from "../../mocks/mocks-test.js";
import {filteredGenre} from "../../mocks/mocks-test.js";

it(`Render GenresList`, () => {
  const tree = renderer.create(<GenresList
    films={films}
    genresList={genresList}
    filteredGenre={filteredGenre}
    onGenreChange={() => {}}
  />, {
    createNodeMock: () => {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
