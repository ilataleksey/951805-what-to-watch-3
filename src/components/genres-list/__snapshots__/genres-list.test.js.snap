// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render GenresList 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      genre-1
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      genre-2
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      genre-3
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      genre-4
    </a>
  </li>
</ul>
`;
